bodo/libs/_bodo_common.cpp

```diff
@@ -142,6 +145,23 @@ Bodo_CTypes::CTypeEnum arrow_to_bodo_type(arrow::Type::type type) {
     }
 }
 
+bodo_array_type::arr_type_enum type_to_array_type(Bodo_CTypes::CTypeEnum typ) {
```

Comment:
@DrTodd13 @srilman what can we do to make these type conversions more complete and robust without a lot of effort?